sycl/doc/extensions/supported/sycl_ext_oneapi_backend_level_zero.md

```diff
@@ -115,6 +115,7 @@ struct {
 ``` C++
 struct {
   ze_command_queue_handle_t NativeHandle;
+  device Device;

```

Comment:
Do we need to mention that the old variant is still usable, although deprecated? Maybe it is covered by:

> NOTE: This extension is following SYCL 2020 backend specification. Prior API for interoperability with Level-Zero is marked as deprecated and will be removed in the next release.

We could include the constructors, but they are really only there to allow `Device` to be before `Ownership` as it is more in line with other input types, like the one for `context`. I fear including the constructors here will only make it less readable.

Reply:
Yes, I think we should document the deprecated API.  Since this PR changes the API, the feature-test macro should also be incremented to `3`, and the new form of `make_queue` should note that it was added in version `3` of this extension.

I don't understand your comment about the constructor.  Are you referring to the `make_queue` function as the "constructor"?

Reply:
> I don't understand your comment about the constructor. Are you referring to the make_queue function as the "constructor"?

By constructor, I mean the constructor of the struct the new `Device` member. It is unnamed in this document, but in the implementation we have

```c++
template <> struct BackendInput<backend::ext_oneapi_level_zero, queue> {
  struct type {
    interop<backend::ext_oneapi_level_zero, queue>::type NativeHandle;
    device Device;
    ext::oneapi::level_zero::ownership Ownership{
        ext::oneapi::level_zero::ownership::transfer};

    type(interop<backend::ext_oneapi_level_zero, queue>::type nativeHandle,
         ext::oneapi::level_zero::ownership ownership) ...

    type(interop<backend::ext_oneapi_level_zero, queue>::type nativeHandle,
         device dev, ext::oneapi::level_zero::ownership ownership) ...
  };
```

This allows both `make_queue(... {ZeCommandQueueHandle, MyOwnership} ...)` and `make_queue(... {ZeCommandQueueHandle, MyDevice, MyOwnership} ...)`, where the former is deprecated.